src/main/scala/guessbook/BookGuessChecker.scala

```diff
@@ -0,0 +1,99 @@
+package guessbook
+
+import guesswho.GenderEnum
+
+trait BookGuessChecker[T] {
+  def validateGuess(guessAttribute: BookAttribute, guessValue: T) : Unit
+  def printFeedback(selectedBook: Book, guessAttribute: BookAttribute, guessValue: T): Unit
+}
+
+object BookGuessChecker {
+  implicit object BooleanGuessChecker extends BookGuessChecker[Boolean] {
+    def validateGuess(guessAttribute: BookAttribute, guessValue: Boolean) : Unit = {
+      guessAttribute match {
+        case Honkaku | SchoolOrUniversityStudents | Series | PushkinVertigo =>
+        case _ => throw new Exception("Invalid attribute for a Boolean guess value")
+      }
+    }
+
+    def printFeedback(selectedBook: Book, guessAttribute: BookAttribute, guessValue: Boolean): Unit = {
+      if (selectedBook.attributes(guessAttribute) == guessValue && guessValue) { // guessed true, actual true
+        println(s"Yes, the selected book is ${guessAttribute.attributeText}")
+      } else if (selectedBook.attributes(guessAttribute) != guessValue && guessValue) { // guessed true, actual false
+        println(s"No, the selected book is not ${guessAttribute.attributeText}")
+      } else if (selectedBook.attributes(guessAttribute) == guessValue && !guessValue) { // guessed false, actual false
+        println(s"Yes, the selected book is not ${guessAttribute.attributeText}")
+      } else { // guessed false, actual true
+        println(s"No, the selected book is ${guessAttribute.attributeText}")
+      }
+    }
+  }
+
+  implicit object GenderGuessChecker extends BookGuessChecker[GenderEnum.Value] {
+    def validateGuess(guessAttribute: BookAttribute, guessValue: GenderEnum.Value) : Unit = {
+      guessAttribute match {
+        case AuthorGender =>
+        case _ => throw new Exception("Invalid attribute for a gender guess value")
+      }
+    }
+
+    def printFeedback(selectedBook: Book, guessAttribute: BookAttribute, guessValue: GenderEnum.Value): Unit = {
+      if (selectedBook.attributes(guessAttribute) == guessValue) {
+        println(s"Yes, the author of the selected book is ${guessValue.toString.toLowerCase}")
+      } else {
+        println(s"No, the author of the selected book is not ${guessValue.toString.toLowerCase}")
+      }
+    }
+  }
+
+  implicit object CountryGuessChecker extends BookGuessChecker[Country.Value] {
+    def validateGuess(guessAttribute: BookAttribute, guessValue: Country.Value) : Unit = {
```

Comment:
is guessValue being used?

Reply:
The type of guessValue determines which version of validateGuess() runs (if only guessAttribute is passed in, the implicit BookGuessChecker won't know which one to use). But I've now also included guessValue in the exception message if the attribute is invalid